### DISCUSSION TITLE
CSV agent with JS!!🎉🎉

### DISCUSSION THREAD

**Discussion by danieljqs1** [FlowiseAI/Flowise#598] [Q&A]
It's great to see the CSV agent added!

Just wanted to ask general indication of how cautious to be with it, I was in the understanding that it worked on pythons pandas library and it wasn't possible to replicate on JS. I would love to use it for a client and wanted to get your thoughts

Did you find a workaround?

Thanks for the response

**Comment by HenryHengZJ** [comment]
I added Pyodide so that it can now use pandas numpy and many other python libraries.
See implementation here - https://github.com/FlowiseAI/Flowise/pull/595

also introduced [airtable agent](https://github.com/FlowiseAI/Flowise/pull/600). both works really well

**Reply by danieljqs1** [reply]
Amazing, thanks for the explanation!

**Reply by gcod-valtech** [reply]
I get the error TypeError: Cannot read properties of undefined (reading 'getRepository') with the Airtable Agent. Any idea why?

**Reply by HenryHengZJ** [reply]
will be fixed in the next release @gcod-valtech !
